lib/DDG/Rewrite.pm

```diff
@@ -135,53 +135,72 @@ sub _build_nginx_conf {
 	my $cfg = "location ^~ ".$self->path." {\n";
 	$cfg .= "\tproxy_set_header Accept '".$self->accept_header."';\n" if $self->accept_header;
 	
-        if($uses_echo_module) {
-            # we need to make sure we have plain text coming back until we have a way
-            # to unilaterally gunzip responses from the upstream since the echo module
-            # will intersperse plaintext with gzip which results in encoding errors.
-            # https://github.com/agentzh/echo-nginx-module/issues/30
-            $cfg .= "\tproxy_set_header Accept-Encoding '';\n";
-
-            # This is a workaround that deals with endpoints that don't support callback functions.
-            # So endpoints that don't support callback functions return a content-type of 'application/json'
-            # because what they're returning is not meant to be executed in the first place.
-            # Setting content-type to application/javascript for those endpoints solves blocking due to 
-            # mime type mismatches.
-            $cfg .= "\tmore_set_headers 'Content-Type: application/javascript; charset=utf-8';\n";
-        }
+	if($uses_echo_module) {
+		# we need to make sure we have plain text coming back until we have a way
+		# to unilaterally gunzip responses from the upstream since the echo module
+		# will intersperse plaintext with gzip which results in encoding errors.
+		# https://github.com/agentzh/echo-nginx-module/issues/30
+		$cfg .= "\tproxy_set_header Accept-Encoding '';\n";
+
+		# This is a workaround that deals with endpoints that don't support callback functions.
+		# So endpoints that don't support callback functions return a content-type of 'application/json'
+		# because what they're returning is not meant to be executed in the first place.
+		# Setting content-type to application/javascript for those endpoints solves blocking due to 
+		# mime type mismatches.
+		$cfg .= "\tmore_set_headers 'Content-Type: application/javascript; charset=utf-8';\n";
+	}
 
 	if($uses_echo_module || $self->accept_header || $is_duckduckgo) {
-	    $cfg .= "\tinclude /usr/local/nginx/conf/nginx_inc_proxy_headers.conf;\n";
+		$cfg .= "\tinclude /usr/local/nginx/conf/nginx_inc_proxy_headers.conf;\n";
 	}
 
 	$cfg .= "\techo_before_body '".$self->callback."(';\n" if $wrap_jsonp_callback;
 	$cfg .= "\techo_before_body '".$self->callback.qq|("';\n| if $wrap_string_callback;
+
+	my $spice_name;
+	if(($spice_name) = $self->path =~ m/js\/spice\/(.+)\/?/){
```

Comment:
Consider using m{} or the likes with paths so you don't need to escape /.  These paths are all of the form,
```
/js/spice/name/of/spice/
```
right?  We should be able to anchor it and skip the substitution on line 162 as well:
```
m{^/js/spice/(.+?)/?$}
```
We probably can replace `/?` with `/` as well since they all seem to have the trailing slash.